app/routes/docs/schema-browser/$version/$.tsx

```diff
@@ -1,34 +1,164 @@
-/*!
- * Copyright © 2022 United States Government as represented by the Administrator
- * of the National Aeronautics and Space Administration. No copyright is claimed
- * in the United States under Title 17, U.S. Code. All Other Rights Reserved.
- *
- * SPDX-License-Identifier: NASA-1.3
- */
 import type { DataFunctionArgs } from '@remix-run/node'
-import { Link, useLoaderData } from '@remix-run/react'
-import { Table } from '@trussworks/react-uswds'
+import { Link, NavLink, useLoaderData } from '@remix-run/react'
+import {
+  CardBody,
+  CardHeader,
+  Icon,
+  SideNav,
+  Table,
+} from '@trussworks/react-uswds'
+import { useState } from 'react'
+import { json, redirect, useParams } from 'react-router'
 
-import { Highlight } from '~/components/Highlight'
-import { Tab, Tabs } from '~/components/Tabs'
-import { loadJson, loadSchemaExamples } from '~/lib/schema-data'
-import type { SchemaProperty } from '~/routes/docs/schema-browser/SchemaBrowserElements.lib'
+import SchemaDefinition from '.'
+import type { Schema } from '../SchemaBrowserElements.lib'
 import {
   ReferencedElementTable,
   SchemaPropertiesTableBody,
-} from '~/routes/docs/schema-browser/SchemaBrowserElements.lib'
-
-export async function loader({ params: { '*': path } }: DataFunctionArgs) {
-  if (!path) throw new Response(null, { status: 404 })
-  if (!path.includes('.schema.json')) throw new Response(null, { status: 404 })
-  const examples = await loadSchemaExamples(path)
-  const result = await loadJson(path)
+  formatFieldName,
+  formatFieldType,
+} from '../SchemaBrowserElements.lib'
+import DetailsDropdownButton from '~/components/DetailsDropdownButton'
+import DetailsDropdownContent from '~/components/DetailsDropdownContent'
+import { Highlight } from '~/components/Highlight'
+import { SideNavSub } from '~/components/SideNav'
+import { Tab, Tabs } from '~/components/Tabs'
+import { publicStaticShortTermCacheControlHeaders } from '~/lib/headers.server'
+import type { ExampleFiles, GitContentDataResponse } from '~/lib/schema-data'
+import {
+  getGithubDir,
+  getVersionRefs,
+  loadJson,
+  loadSchemaExamples,
+} from '~/lib/schema-data'
 
-  return { path, result, examples }
+export async function loader({
+  params: { version, '*': path },
+}: DataFunctionArgs) {
+  if (!version) throw new Response(null, { status: 404 })
+  if (path?.endsWith('/')) {
+    return redirect(`${path.slice(0, -1)}`)
+  }
```

Comment:
Why is this necessary?

Reply:
Ah maybe not totally needed, there were errors when the path ended with '/' because of how it tried to get the content data from the api. We could just let it throw the error

Reply:
I guess it's fine.